src/app/js/lib/DialogButton.js

```diff
@@ -0,0 +1,108 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import translate from 'redux-polyglot/translate';
+import compose from 'recompose/compose';
+import withHandlers from 'recompose/withHandlers';
+import { submit as submitAction } from 'redux-form';
+import FlatButton from 'material-ui/FlatButton';
+import Dialog from 'material-ui/Dialog';
+import IconButton from 'material-ui/IconButton';
+import classnames from 'classnames';
+import CircularProgress from 'material-ui/CircularProgress';
+
+import ButtonWithStatus from './ButtonWithStatus';
+import { polyglot as polyglotPropTypes } from '../propTypes';
+
+export const DialogButtonComponent = ({
```

Comment:
I would rename this component to `ButtonWithDialog`. I first thought that it was a button for inside the dialog